src/UI/Components/InstanceWithRelationsProvider/InstanceWithRelationsProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useContext } from "react";
+import {
+  InstanceWithRelations,
+  useGetInstanceWithRelations,
+} from "@/Data/Managers/V2/GetInstanceWithRelations";
+import { DependencyContext, words } from "@/UI";
+import { ErrorView, LoadingView } from "@/UI/Components";
+
+/**
+ * Renders the InstanceWithRelationsProviderProvider component.
+ * It serves purpose to extract the data fetching logic and provide the required instance data for the Dependent
```

Comment:
I really think the composer could benefit using a context instead of needing to pass on data through each level. You can look at how I handled it with the service instance details, it helps keeping your code simple and avoids the need for overly nested Dependencies.
You'd have less rerender issues too.
I get that this is open for debate and will conflict with the ongoing work on the other sidebar.

Reply:
added composer